### PR TITLE
chore: Single UI Provider

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR joins duplicated contexts we had on our code. 

## How does it work?
Both `UIProvider` and `ModalProvider` had a goal of controlling the global UI state of our site. This created an interdependency between these contexts, generating a mess on our code. For example, for openning the cart on UIProvider we relied on functions provided by `ModalProvider`. Also, when openning the cart we had to invoke a function called `onModalOpen`, what makes no sense since the cart is a drawer, not a modal.

This PR addresses this issue by merging UIProvider and ModalProvider into a single context so we don't have that mess anymore. Also, I have changed the name of `onModalClose` and `onModalOpen` to `fadeOut` and `fadeIn` respectively. 

After these changes, the minicart toggle button went from:
```
const { closeCart } = useUI()
const { onModalClose } = useModal()

const onClick = () => {
    onModalClose()
    closeCart()
}

``` 

to: 
```
const { closeCart, fadeOut } = useUI()

const onClick = () => {
    fadeOut()
    closeCart()
}
```

## How to test it?
Make sure all fade effects are working correctly, as well as the scroll behavior when the modal is open

### CODE REVIEW

File: src/components/search/Filter/Filter.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ function Filter({
     state: { selectedFacets },
   } = useSearch()
 
-  const { onModalClose } = useModal()
+  const { closeModal } = useUI()
```
Comment (eduardoformiga) [inline src/components/search/Filter/Filter.tsx:43]: should we create a new state for the `Filter` on the provider? with `openFilter`, `closeFilter` etc?

**Review by eduardoformiga — COMMENTED** [review]
The Provider merge was awesome! 👏 

I noticed some problems with the `Filter`, and the scroll, though.

I think we need to tweak the `no-scroll` logic and maybe create a new state for the filter on the Provider. What do you think?

|Mobile|Desktop|
|-|-|
|gyS3pnAvb6|tlE1WtOL75|

File: src/components/search/Filter/Filter.tsx (hunk 1 above)
Comment (tlgimenes) [reply src/components/search/Filter/Filter.tsx]: great idea! I'll do it

File: src/components/common/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -129,43 +168,7 @@ function Navbar() {
...
-              <div className="navlinks__signin">
-                <Suspense fallback={<ButtonSignInFallback />}>
-                  <ButtonSignIn />
-                </Suspense>
-              </div>
-            </div>
-          </div>
-        </SlideOver>
-      )}
+      {displayNavbar && <NavbarSlider />}
```
Comment (eduardoformiga) [inline src/components/common/Navbar/Navbar.tsx:171]: Nice!!

**Review by eduardoformiga — APPROVED** [review]
Great job! ✨ 
Everything looks more readable now.

I noticed some SonarQube complains while reviewing.
Screen Shot 2022-06-08 at 13 19 54

Another little thing but I'm not sure if it's related to the PR.
The facets accordion item is collapsed after opening the second time on:

NDjJCE5bad

File: src/components/common/Navbar/Navbar.tsx (hunk 2 above)
Comment (tlgimenes) [reply src/components/common/Navbar/Navbar.tsx:171]: Thanks! yes! those issues on the filter were related to this PR! thanks for the awesome review!

**Pull request merged**